app/src/main/java/br/com/dionataferraz/vendas/login/LoginActivity.kt

```diff
@@ -43,28 +68,23 @@ class LoginActivity : AppCompatActivity() {
                     }
                 }
                 val startIndexOfLink = this.text.toString().indexOf(link.first)
-                spannableString.setSpan(clickableSpan, startIndexOfLink, startIndexOfLink + link.first.length,
-                    Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
+                spannableString.setSpan(
+                    clickableSpan, startIndexOfLink, startIndexOfLink + link.first.length,
+                    Spanned.SPAN_EXCLUSIVE_EXCLUSIVE
+                )
             }
             this.movementMethod = LinkMovementMethod.getInstance()
             this.setText(spannableString, TextView.BufferType.SPANNABLE)
         }
 
         binding.btRegistro.makeLinks(
-            Pair("Registrar", View.OnClickListener {
-                val intent  = Intent(this, ProfileActivity::class.java)
-                startActivity(intent)
-            })
+                Pair("Registrar", View.OnClickListener {
+                    Intent(this, ProfileActivity::class.java).also {
+                        startActivity(it)
+                        finish()
+                    }
+                }
+            )
         )
```

Comment:
Não entendi o Pair, mas só o setOncliclistener já resolveria
```suggestion
binding.btRegistro.setOncliclistener{
                    Intent(this, ProfileActivity::class.java).also {
                        startActivity(it)
                        finish()
} 
```